packages/agents/src/agents/Chameleon/StandardRagBuilder.ts

```diff
@@ -0,0 +1,54 @@
+import { OpenAIEmbeddingAPI, LocalVectorDB } from "@evo-ninja/agent-utils";
+import { AgentContext } from "../../AgentContext";
+
+export class StandardRagBuilder<TItem> {
+  private _limit: number;
+  private _items: TItem[];
+  private _selector: (item: TItem) => string;
+
+  constructor(private readonly context: AgentContext) {
+  }
+
+  items(items: TItem[]): StandardRagBuilder<TItem> {
+    this._items = items;
+    return this;
+  }
+
+  limit(limit: number): StandardRagBuilder<TItem> {
+    this._limit = limit;
+    return this;
+  }
+
+  selector(selector: (item: any) => string): StandardRagBuilder<TItem> {
+    this._selector = selector;
+    return this;
+  }
+
+  async query(query: string): Promise<TItem[]> {
+    const embeddingApi = new OpenAIEmbeddingAPI(
+      this.context.env.OPENAI_API_KEY,
+      this.context.logger,
+      this.context.chat.tokenizer
+    );
+
+    const db = new LocalVectorDB(this.context.workspace, "testdb", embeddingApi);
+
+    const uuid = Math.floor(Date.now() / 1000).toString(16);
+
+    const collection = db.addCollection(uuid);
```

Comment:
Why are we creating a new collection for each query? It would be nice if we could reuse a collection that already contains the items. This way subsequent queries wouldn't need to re-add all items.

Reply:
Would be nice ofc, but optimizations are the last thing on my mind right now